docs/decex/fraud_proofs.md

```diff
@@ -368,6 +368,12 @@ Proving and verification algorithm varies depending on which execution phase the
 4. If the post-state root matches the state root in the trace of the header being challenged, then the proof is invalid ⇒ ignore the proof
 5. If the post-state root does not match the state root in the trace, then the proof is valid ⇒ accept the proof and punish the authors of receipt identified by `bad_receipt_hash`.
 
+## Fraud proof priority in transaction pool
+
+For fraud proof that targets bad ER its priority is defined as `MAX - blocks_before_bad_er_confirm`, fraud proof targets bad ER that is closer to being confirmed is more emergency and thus has a higher priority to be accepted by the transaction pool and to be included in the next block. For a given domain, at most one fraud proof will be accepted by the transaction pool at a time, if an incoming fraud proof has a higher priority than the fraud proof already in the pool then it will replace the previous fraud proof otherwise it will be rejected.
```

Comment:
```suggestion
To prune a chain of fraudulent Execution Receipts and slash the operators who submitted them, it is sufficient to include in a consensus block and process only a single fraud proof for the oldest fraudulent ER. If the fraud proof is valid, all the children blocks will be pruned and operators slashed automatically.
A fraud proof that targets a bad ER has a priority is defined as `MAX - blocks_before_bad_er_confirm`, where `blocks_before_bad_er_confirm` is how many blocks remain until this ER is outside the challenge period. A fraud proof that targets a bad ER that is closer to being confirmed is more urgent and thus has a higher priority to be accepted by the transaction pool and to be included in the next consensus block. For a given domain, at most one fraud proof will be accepted by the transaction pool at a time; if an incoming fraud proof has a higher priority than the fraud proof already in the pool then it will replace the previous fraud proof, otherwise it will be rejected.
```